src/Appwrite/Extend/Exception.php

```diff
@@ -0,0 +1,186 @@
+    const STORAGE_INVALID_FILE_SIZE         = 'storage_invalid_file_size';
+    const STORAGE_INVALID_FILE              = 'storage_invalid_file';
+    const STORAGE_FAILED_TO_MOVE_FILE       = 'storage_failed_to_move_file';
+    const STORAGE_FAILED_TO_WRITE_FILE      = 'storage_failed_to_write_file';
+
+    /** Functions */
+    const FUNCTION_NOT_FOUND       = 'function_not_found';
+    const FUNCTION_DELETION_FAILED = 'function_deletion_failed';
+
+    /** Deployments */
+    const DEPLOYMENT_NOT_FOUND       = 'deployment_not_found';
+    const DEPLOYMENT_DELETION_FAILED = 'deployment_deletion_failed';
+
+    /** Execution */
+    const EXECUTION_NOT_FOUND       = 'execution_not_found';
+
+    /** Collections */
+    const COLLECTION_NOT_FOUND       = 'collection_not_found';
+    const COLLECTION_ALREADY_EXISTS  = 'collection_already_exists';
+    const COLLECTION_LIMIT_EXCEEDED  = 'collection_limit_exceeded';
+    const COLLECTION_DELETION_FAILED = 'collection_deletion_failed';
+    
+    /** Documents */
+    const DOCUMENT_NOT_FOUND         = 'document_not_found';
+    const DOCUMENT_INVALID_STRUCTURE = 'document_invalid_structure';
+    const DOCUMENT_MISSING_PAYLOAD   = 'document_missing_payload';
+    const DOCUMENT_ALREADY_EXISTS    = 'document_already_exists';
+
+    /** Attribute */
+    const ATTRIBUTE_NOT_FOUND           = 'attribute_not_found';
+    const ATTRIBUTE_UNKNOWN             = 'attribute_unknown';
+    const ATTRIBUTE_NOT_AVAILABLE       = 'attribute_not_available';
+    const ATTRIBUTE_FORMAT_UNSUPPORTED  = 'attribute_format_unsupported';
+    const ATTRIBUTE_DEFAULT_UNSUPPORTED = 'attribute_default_unsupported';
+    const ATTRIBUTE_ALREADY_EXISTS      = 'attribute_already_exists';
+    const ATTRIBUTE_LIMIT_EXCEEDED      = 'attribute_limit_exceeded';
+    const ATTRIBUTE_VALUE_INVALID       = 'attribute_value_invalid';
+
+    /** Indexes */
+    const INDEX_NOT_FOUND       = 'index_not_found';
+    const INDEX_LIMIT_EXCEEDED  = 'index_limit_exceeded';
+    const INDEX_ALREADY_EXISTS  = 'index_already_exists';
+
+    /** Query limit exceeded */
+    const QUERY_LIMIT_EXCEEDED = 'query_limit_exceeded';
+    const QUERY_INVALID        = 'query_invalid';
+
+    /** Projects */
+    const PROJECT_NOT_FOUND       = 'project_not_found';
+    const PROJECT_UNKNOWN         = 'project_unknown';
+
+    /** Webhooks */
+    const WEBHOOK_NOT_FOUND       = 'webhook_not_found';
+
+    /** Keys */
+    const KEY_NOT_FOUND       = 'key_not_found';
+
+    /** Platform */
+    const PLATFORM_NOT_FOUND       = 'platform_not_found';
+
+    /** Domain */
+    const DOMAIN_NOT_FOUND           = 'domain_not_found';
+    const DOMAIN_ALREADY_EXISTS      = 'domain_already_exists';
+    const DOMAIN_UNREACHABLE         = 'domain_unreachable';
+    const DOMAIN_VERIFICATION_FAILED = 'domain_verification_failed';
+
+    /** General */
+    const DEFAULT                   = 'default';
```

Comment:
Can we add a prefix for general errors as well?

Reply:
done 👍

Reply:
done 👍